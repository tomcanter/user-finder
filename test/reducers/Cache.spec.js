import cacheReducer from 'reducers/Cache';

describe('Reducer: cache', () => {

  describe('when passed no state', () => {
    it('should return the initial state', () => {
      expect(
        cacheReducer(undefined, {})
      ).toMatchSnapshot();
    });
  });

  describe('search cache', () => {

    describe('when a cache entry does not exist', () => {
      it('should add an entry to the cache', () => {
        const beforeState = {
          search: {},
        };
        const action = {
          type: 'SEARCH_SUCCESS',
          query: 'page=2',
          result: [1,2],
          totalResults: 2,
        };
        expect(cacheReducer(beforeState, action)).toMatchSnapshot();
      });
    });

    describe('when a cache entry does exist', () => {
      it('should just return the state', () => {
        const beforeState = {
          search: {
            'page=2': {someProp: 'test'},
          },
        };
        const action = {
          type: 'SEARCH_SUCCESS',
          query: 'page=2',
          result: [1,2],
        };
        expect(cacheReducer(beforeState, action)).toMatchSnapshot();
      });
    });

  });

});
