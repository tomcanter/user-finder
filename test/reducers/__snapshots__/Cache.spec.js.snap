// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: cache search cache when a cache entry does exist should just return the state 1`] = `
Object {
  "search": Object {
    "page=2": Object {
      "someProp": "test",
    },
  },
}
`;

exports[`Reducer: cache search cache when a cache entry does not exist should add an entry to the cache 1`] = `
Object {
  "search": Object {
    "page=2": Object {
      "result": Array [
        1,
        2,
      ],
      "totalResults": 2,
    },
  },
}
`;

exports[`Reducer: cache when passed no state should return the initial state 1`] = `
Object {
  "search": Object {},
}
`;
